include/psa/crypto_extra.h

```diff
@@ -112,10 +112,9 @@ void mbedtls_psa_crypto_free( void );
  * This is an Mbed TLS extension.
  *
  * \note This function is only available on the following platforms:
- * * If the compile-time options MBEDTLS_ENTROPY_NV_SEED and
- *   MBEDTLS_PSA_CRYPTO_STORAGE_C are both enabled. Note that you
- *   must provide compatible implementations of mbedtls_nv_seed_read
- *   and mbedtls_nv_seed_write.
+ * * If the compile-time option MBEDTLS_PSA_INJECT_ENTROPY is enabled.
+ *   Note that you must provide compatible implementations of
+ *   mbedtls_nv_seed_read and mbedtls_nv_seed_write.
```

Comment:
This looks like a breaking change to how folks were configuring the library before (e.g. in Mbed OS). What's the migration path? Any way we can make it a (potentially deprecated) no-op on their part?

Should we have a backwards-compatibility options where if both `MBEDTLS_ENTROPY_NV_SEED` and `MBEDTLS_PSA_CRYPTO_STORAGE_C` are defined, we define `MBEDTLS_PSA_INJECT_ENTROPY` automatically?

Reply:
The migration path is to change the configuration in Mbed OS. We've never promised to maintain backward compatibility with the service/library API.

Entropy injection is a delicate feature and really should be enabled explicitly at the moment. We should make it more robust, and then make it automatic if the prerequisites are available (`MBEDTLS_ENTROPY_NV_SEED` and `MBEDTLS_PSA_CRYPTO_STORAGE_C`), but at the moment it only works if there are no actual entropy sources, only the NV seed.